.github/workflows/security_dismissal.yml

```diff
@@ -0,0 +1,136 @@
+name: Track Dismissed Code Scanning Alerts
+
+on:
+  push:
+    branches: 
+     - dismissalCheckout   # Trigger only test branch, remember to remove before PRing to grass
+                           # in the grass repo this will be changed to a scheduled event (e.g., every 24 hours)
+
+jobs:
+    get-dismissals-data:
+        runs-on: ubuntu-latest  
+        permissions: write-all
+
+        steps:
+     
+             # will remove this step once some testing has been done to make sure it is unnecessary
+           - uses: actions/checkout@v4 # Checkout repository code
+    
+           - name: Install dependencies (if needed for parsing security tool response)
+             run: |
+                # Add any package installation commands here (e.g., pip install requests)
+                  sudo apt-get install jq  # Install jq for parsing JSON
+                 
+   
+           - name: Get Security Alerts
+             run: |
+
+               #get only dismissed alerts using curl
+               # path to local repo will be changed to the path to the repo in grass
+
+               curl -L \
+                -H "Accept: application/vnd.github+json" \
+                -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                https://api.github.com/repos/ncsu-csc472-spring2024/grass-CI-playground/code-scanning/alerts?state=dismissed > alerts.json
+                echo "alerts retrieved successfully!"
+
+
+         
+
+             #trigger a rerun
+           - name: Upload the coding alerts as artifact
+             uses: actions/upload-artifact@v4
+             with:
+              name: workflow-data
+              path: alerts.json
+
+
+             # get the repository issues 
+           - name: Get Current Issues        
+             run: |
+              
+               curl -L \
+               -H "Accept: application/vnd.github+json" \
+               -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+               -H "X-GitHub-Api-Version: 2022-11-28" \
+               https://api.github.com/repos/ncsu-csc472-spring2024/grass-CI-playground/issues > issues.json  
+ 
```

Comment:
Did you ever try the actions/github-script to write JavaScript code with already authenticated object for api calls? Or is there a way that gh cli provides already authenticated way to do calls?

Reply:
The first bullet point of the README https://github.com/actions/github-script?tab=readme-ov-file#actionsgithub-script

An example of where JavaScript and json rest goes well and works hand in hand:
https://github.com/actions/github-script?tab=readme-ov-file#comment-on-an-issue